app/src/main/java/com/example/parcialtp3/ui/view/MainActivity.kt

```diff
@@ -28,10 +33,24 @@ class MainActivity : AppCompatActivity() {
             supportFragmentManager.findFragmentById(R.id.fragmentContainerView) as NavHostFragment
         navHostController = navHostFragment.navController
         bottomNavView = findViewById(R.id.bottom_bar)
-
         NavigationUI.setupWithNavController(bottomNavView, navHostController)
+        setupNavHeader()
         setupDrawerLayout()
     }
+
+    private fun setupNavHeader() {
+        val userNameTextView =
+            navigationView.getHeaderView(0).findViewById<TextView>(R.id.txt_UserName)
+        var loginViewModel = ViewModelProvider(this).get(LoginViewModel::class.java)
+        val userNameObserver: Observer<String?> = object : Observer<String?> {
+            override fun onChanged(userName: String?) {
+                userNameTextView.text = userName
+            }
+        }
+        loginViewModel.userName.observe(this, userNameObserver);
```

Comment:
Intentaron hacer esto ?
loginViewModel.userName.observe(viewLifecycleOwner) { result ->
            userNameTextView.text = result.toString()
        }

Es mas corto y limpio

Reply:
Ahi lo probe y anda. Solo habria que cambiar el viewLifecycleOwner por this, ya que estamos ejecutando esto en el main activity